Expensify/App
# Fix emoji search behavior when deleting all text

### Details
When all the text is deleted from the search, the highlighted emoji index was being reset to `8` (number of columns) instead of `-1` (no emoji selected). This behavior was causing bugs like duplicating emoji section headers as well as inserting the text `frequentlyUsed` into the action compose input.

### Fixed Issues
$ https://github.com/Expensify/App/issues/8824

### Tests and QA
NOTE: If you find yourself with multiple emoji section headers, there is only one way to fix them:
1. Log into that account on https://expensify.com (OldDot)
2. Run `NVP.set('expensify_frequentlyUsedEmojis', []);` from the JS console

To test that this fix is working properly:
1. Open a chat
2. Click the emoji picker to open it
3. Type any text into the search box
4. Delete all the text in the search box
5. Press the Enter key
6. Verify that nothing happens (there is nothing inserted into the action compose input)
7. Verify that you can open and close the emoji picker and the emoji sections are not duplicated
8. Open the emoji picker again
9. Select any emoji
10. Verify the emoji popover closes, the emoji was added to the compose input
11. Open the emoji picker
12. Verify that the emoji was added to the section of frequently used emoji


### QA Steps
Same as tests

### Screenshots


#### Web
Imagine me pressing enter repeatedly at the end of this video

https://user-images.githubusercontent.com/1228807/165869858-89864ea3-a6eb-499c-a145-6b073b9f8640.mp4


#### Mobile Web
Does not apply because there is no search box

#### Desktop
Imagine me pressing enter repeatedly at the end of this video

https://user-images.githubusercontent.com/1228807/165870032-ac59ba05-a983-4807-b274-1c54d315d471.mp4


#### iOS
Does not apply because there is no search box

#### Android
Does not apply because there is no search box
